tests/src/runtimeApi/stream/hipStreamAddCallback.cpp

```diff
@@ -23,41 +23,65 @@ THE SOFTWARE.
  * HIT_END
  */
 
-// Test under-development. Call hipStreamAddCallback function and see if it works as expected.
-
+#include <stdio.h>
 #include "hip/hip_runtime.h"
 #include "test_common.h"
 
 #ifdef __HIP_PLATFORM_HCC__ 
 #define HIPRT_CB
 #endif
 
-class CallbackClass
+__global__ void vector_square(float *C_d, float *A_d, size_t N)
 {
-public:
-    static void HIPRT_CB Callback(hipStream_t stream, hipError_t status, void *userData);
+    size_t offset = (blockIdx.x * blockDim.x + threadIdx.x);
+    size_t stride = blockDim.x * gridDim.x ;
+
+    for (size_t i=offset; i<N; i+=stride) {
+        C_d[i] = A_d[i] * A_d[i];
+    }
+}
 
-private:
-    void callbackFunc(hipError_t status);
-};
+float *A_h, *C_h;
+size_t N = 1000000;
 
-void HIPRT_CB CallbackClass::Callback(hipStream_t stream, hipError_t status, void *userData)
+static void HIPRT_CB Callback(hipStream_t stream, hipError_t status, void *userData)
 {
-    CallbackClass* obj = (CallbackClass*) userData;
-    obj->callbackFunc(status);
+    for (size_t i=0; i<N; i++) {
```

Comment:
++i

Reply:
Array index starts at 0. So i++ is correct.

Reply:
It is correct, but not because of index starting value - it doesn't matter. i++ reads the value of i then increments it. ++i increments the value of i then reads it. It is about performance, not correctness. Correctness is in case like following: for(int i=0, a = 0; i<10; a = i++), where result of iteration is used in place. Here i++ means increment, but evaluate to the previous value, but ++i means increment, but evaluate to the next value. As a result a will get different values (i when i++ and i+1 when ++i).

Reply:
This is test code. And the for loop is for comparing the results generated by host and device. So it is not a performance critical piece.